CMakeLists.txt

```diff
@@ -1,63 +1,46 @@
 PROJECT(pblib)
 
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required(VERSION 3.5)
 
-# SET (CMAKE_CXX_COMPILER "clang++")
- SET(CMAKE_CXX_COMPILER "g++")
+# options
+option(PBLIB_OSX_VIRTUAL_DESTRUCTOR_WORKAROUND "Hotfix for an older versions of Clang on OS-X" OFF)
 
-set(CMAKE_CXX_FLAGS "-O3 -std=c++11")
-# set(CMAKE_CXX_FLAGS "-O3 -std=c++11 -static")
-# set(CMAKE_CXX_FLAGS "-O3 -g3 -std=c++11")
-# set(CMAKE_CXX_FLAGS "-O1 -g3 -std=c++11")
-# set(CMAKE_CXX_FLAGS "-O3 -g3 -std=c++0x")
+# Force C++11, also disable GNU extensions
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
 
-# set (CMAKE_BUILD_TYPE "Debug")
- set (CMAKE_BUILD_TYPE "Release")
-
-ADD_CUSTOM_TARGET(cpfiles ALL
-   COMMAND mkdir -p lib/encoder &&  cp *.h lib/ && cp encoder/*.h lib/encoder/ && cp libpblib.* lib/
-)
-
-
-file(GLOB pblib_src_files
-    "*.cpp", "encoder/*.cpp"
-)
-
-list(REMOVE_ITEM  pblib_src_files "${pblib_SOURCE_DIR}/fuzzer.cpp" "${pblib_SOURCE_DIR}/pbencoder.cpp")
+# load the MiniSat library
+add_subdirectory(BasicPBSolver/minisat EXCLUDE_FROM_ALL)
 
+# pblib
+file(GLOB pblib_src_files "*.cpp" "encoder/*.cpp")
```

Comment:
GLOB is not supported in all CMake targets, so this will be done at configuration time and will not be redone if new files are added. The files should be listed explicitly instead.